maa-cli/src/run/copilot.rs

```diff
@@ -0,0 +1,204 @@
+use super::run;
+use crate::{
+    config::task::{
+        default_variants, task_type::TaskType, value::input::Input, Strategy, Task, TaskList, Value,
+    },
+    dirs::Dirs,
+    installer::maa_core::find_resource,
+    normal, object, warning,
+};
+use anyhow::anyhow;
+use anyhow::Context;
+use anyhow::Result;
+use prettytable::{format, row, Table};
+use reqwest::blocking::Client;
+use serde_json::Value as JsonValue;
+use std::{
+    fs::{self, File},
+    io::{self, BufReader, Write},
+    path::{Path, PathBuf},
+};
+
+pub fn copilot(
+    dirs: &Dirs,
+    link: String,
+    addr: Option<String>,
+    user_resource: bool,
+    batch: bool,
+    dry_run: bool,
+) -> Result<()> {
+    let api = "https://prts.maa.plus/copilot/get/";
+    let cache_dir = dirs.cache().display().to_string();
+    let mut stage_dir = match find_resource(dirs).context("Failed to find resource!") {
+        Ok(stage_dir) => stage_dir,
+        Err(_) => return Err(anyhow!("Resource not found")),
+    };
+    stage_dir.push("Arknights-Tile-Pos");
+
+    let copilot_id = match link.get(link.len() - 5..) {
+        Some(id) => id,
+        None => {
+            return Err(anyhow!("Code Invaild"));
+        }
+    };
```

Comment:
This can be simplified by `.context`:
```suggestion
    let copilot_id = link.get(link.len() - 5..).context("Code Invalid")?;
```